java/util/src/main/java/com/google/protobuf/util/JsonFormat.java

```diff
@@ -1706,6 +1724,10 @@ private EnumValueDescriptor parseEnum(EnumDescriptor enumDescriptor, JsonElement
           // Fall through. This exception is about invalid int32 value we get from parseInt32() but
           // that's not the exception we want the user to see. Since result == null, we will throw
           // an exception later.
+
+          if (usingUnrecognizedEnumValue) {
+            result = enumDescriptor.createUnrecognizedValueDescriptor();
```

Comment:
The UNRECOGNIZED value should not be used here. Only generated getters should ever return UNRECOGNIZED and the value should not be stored anywhere. Can you treat it as if the field is not present?

Reply:
Hi @veerenmandalia you had a chance to do this, or I'll be glad to pick it up...

Reply:
I needed the same thing so have created a PR https://github.com/google/protobuf/pull/4825/commits/c26cf191f62e617942f28111fa813f67535c6feb

Please review.